tests/eos-test-convenience.c

```diff
@@ -0,0 +1,202 @@
+  g_assert_nonnull (fixture);
+
+  data->fixture = fixture;
+  data->server = NULL;
+  data->subserver = NULL;
+  data->client = NULL;
+}
+
+void
+etc_data_clear (EtcData *data)
+{
+  if (data == NULL)
+    return;
+
+  data->fixture = NULL;
+  g_clear_object (&data->server);
+  g_clear_object (&data->subserver);
+  g_clear_object (&data->client);
+}
+
+void
+etc_set_up_server (EtcData *data)
+{
+  g_autoptr(GFile) server_root = NULL;
+  g_autofree gchar *keyid = NULL;
+  g_autoptr(GError) error = NULL;
+
+  g_assert_nonnull (data);
+  g_assert_nonnull (data->fixture);
+  g_assert_null (data->server);
+  g_assert_null (data->subserver);
+
+  server_root = g_file_get_child (data->fixture->tmpdir, "main");
+  keyid = get_keyid (data->fixture->gpg_home);
+  data->server = eos_test_server_new_quick (server_root,
+                                            default_vendor,
+                                            default_product,
+                                            default_ref,
+                                            0,
+                                            data->fixture->gpg_home,
+                                            keyid,
+                                            default_ostree_path,
+                                            &error);
+  g_assert_no_error (error);
+  g_assert_cmpuint (data->server->subservers->len, ==, 1u);
+
+  data->subserver = g_object_ref (EOS_TEST_SUBSERVER (g_ptr_array_index (data->server->subservers, 0)));
+}
+
+void
+etc_set_up_client_synced_to_server (EtcData *data)
+{
+  g_autoptr(GFile) client_root = NULL;
+  g_autoptr(GError) error = NULL;
+
+  g_assert_nonnull (data);
+  g_assert_nonnull (data->fixture);
+  g_assert_nonnull (data->server);
+  g_assert_nonnull (data->subserver);
+  g_assert_null (data->client);
+
+  client_root = g_file_get_child (data->fixture->tmpdir, "client");
+  data->client = eos_test_client_new (client_root,
+                                      default_remote_name,
+                                      data->subserver,
+                                      default_ref,
+                                      default_vendor,
+                                      default_product,
+                                      &error);
+  g_assert_no_error (error);
+}
+
+void
+etc_update_server (EtcData *data,
+                   guint commit)
+{
+  g_autoptr(GError) error = NULL;
+  guint current_commit;
+
+  g_assert_nonnull (data);
+  g_assert_nonnull (data->subserver);
+
+  current_commit = GPOINTER_TO_UINT (g_hash_table_lookup (data->subserver->ref_to_commit, default_ref));
+  g_assert_cmpint (current_commit, <, commit);
+
+  g_hash_table_insert (data->subserver->ref_to_commit,
+                       g_strdup (default_ref),
+                       GUINT_TO_POINTER (commit));
+  eos_test_subserver_update (data->subserver,
+                             &error);
+  g_assert_no_error (error);
+}
+
+void
+etc_update_client (EtcData *data)
```

Comment:
It would be useful to have a brief doc comment for each of these utility functions, to clarify what they do. Doesn’t have to be gtk-doc formatted.